From 3b823b9d4f3b19c880e05a71bd16f4c207450d8a Mon Sep 17 00:00:00 2001
From: David Dufresne <david.dufresne@numigi.net>
Date: Thu, 17 Jan 2019 15:45:19 -0500
Subject: [PATCH 1/1] Enable a specific config file

---
 odoo/tools/config.py | 24 +++++++++++++++++++++++-
 1 file changed, 23 insertions(+), 1 deletion(-)

diff --git a/odoo/tools/config.py b/odoo/tools/config.py
index 47c5b633cff..49b2b6a19ce 100644
--- a/odoo/tools/config.py
+++ b/odoo/tools/config.py
@@ -661,4 +661,26 @@ class configmanager(object):
                 self.options['admin_passwd'] = updated_hash
             return True
 
-config = configmanager()
+class ConfigManagerWithSpecificConfigFile(configmanager):
+
+    def load(self):
+        """Load the config from a generic and a specific rc file.
+
+        The generic rc file is the standard Odoo rc file.
+        The specific rc file is given through the environment variable SPECIFIC_ODOO_RC.
+
+        The config is first loaded with the generic file, through the call to super.
+
+        If a specific rc file is given, the super method is called a second time
+        but with the rcfile variable pointing to the specific file.
+        """
+        super().load()
+        generic_rcfile = self.rcfile
+        specific_rcfile = os.environ.get('SPECIFIC_ODOO_RC')
+
+        if specific_rcfile:
+            self.rcfile = specific_rcfile
+            super().load()
+            self.rcfile = generic_rcfile
+
+config = ConfigManagerWithSpecificConfigFile()
-- 
2.17.1

