diff --git a/odoo/modules/registry.py b/odoo/modules/registry.py
index fd94d78c61df..1a589c76fcbd 100644
--- a/odoo/modules/registry.py
+++ b/odoo/modules/registry.py
@@ -75,6 +75,7 @@ class Registry(Mapping):
             with odoo.api.Environment.manage():
                 registry = object.__new__(cls)
                 registry.init(db_name)
+                registry.new = registry.init = registry.registries = None
 
                 # Initializing a registry will call general code which will in
                 # turn call Registry() to obtain the registry being initialized.
@@ -103,7 +104,6 @@ class Registry(Mapping):
             registry._init = False
             registry.ready = True
             registry.registry_invalidated = bool(update_module)
-            registry.new = registry.init = registry.registries = None
 
         return registry
 
